### PR TITLE
[DOCS-5601] Triggers

### PR DESCRIPTION
**Pull request by m-hulbert**

### CODE REVIEW

File: docs/modeling/connectors/README.md [hunk 1]
```diff
@@ -2,61 +2,158 @@
...
+
+Each action has a set of input parameters and output parameters. Input parameters are the values sent from a process to the connector and output parameters are the ones returned to the process after the connector has executed its logic. Values are defined in a service task using [process variables](../processes/variables.md).
+
+The properties for input and output parameters are:
+
+| Property | Description | Example | Required? | 
+| -------- | ----------- | ------- | --------- |
+| `Name` | The name of the parameter | userId | Yes |
+| `Description` | A free text description of the parameter | The ID of a Slack user | No | 
+| `Type` | The data type of the parameter | Array | Yes |
```
Comment (pmartinezga) [inline docs/modeling/connectors/README.md]: `Array` is not a valid parameter type. At the moment the supported types are: string, integer, boolean, date, datetime, file and JSON.

In this concrete case, to follow with the `userId` example, this should be `string`

File: docs/modeling/connectors/README.md [hunk 2]
```diff
@@ -2,61 +2,158 @@
...
+                    "type": "string"
+                },
+...
+            ],
+            "outputs": [
+                {
+                    "id": "fa6b0f2a-ca79-476b-ba4e-f0f082eff47c",
+                    "name": "matchGroups",
+                    "description": "Matching groups between pattern and message. They can be used in variables and the echo messages.",
+                    "type": "array"
```
Comment (pmartinezga) [inline docs/modeling/connectors/README.md]: This type was an error, it should be `json` there is a ticket in JIRA to fix it https://issues.alfresco.com/jira/browse/AAE-1711

File: docs/modeling/triggers.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+| `Description` | A free text description of what the trigger does | A trigger to start a process when an approval email is received. | 
+
+### Events
+Events are created when the configured event criteria are met. 
+The properties for trigger events are: 
+
+| Property | Description | Example | 
+| -------- | ----------- | ------- | 
+| `Connector` | The [connector](../modeling/connectors/README.md) instance to use for an event | email-connector-1 | 
+| `Event` | The [connector event](../modeling/connectors/README.md#events) to use for this trigger | EMAIL_RECEIVED |
```
Comment (pmartinezga) [inline docs/modeling/triggers.md]: The `connector` and the `event` are not properties by itself, they are part of the `source` property.

In fact, in the next version of the triggers, there may be sources like: `rb.rb-processes-events.PROCESS_START` or `rb.rb-tasks-events.TASK_ASSIGNED` so the wat the source is created will be different than the current one in which a connector and an event are selected.

File: docs/modeling/triggers.md [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+| Property | Description | Example | Required | 
+| -------- | ----------- | ------- | -------- |
+| `processDefinitionKey` | The ID of the process to start | Process_YjqRw3n2 | Yes |
+| `name` | The name of the process instance | approval-process | Yes | 
+| `businessKey` | An optional business key ID for the process instance | 14240 | No | 
+| `variables` | Values from the trigger can be mapped to [process variables](../modeling/processes/variables.md). For a connector this will include the output parameters configured in the [connector event](../modeling/connectors/README.md#events) definition. | | No | 
+| `source` | The source is automatically set and is the channel that the action is sent to | command-consumer | 
+
+Actions are stored as a `payload`, for example using the start a process action for the `EMAIL_RECEIVED` event of the [email connector](../modeling/connectors/ootb/email.md):
```
Comment (pmartinezga) [inline docs/modeling/triggers.md:67]: I think it is worth to mention (I don't know where should be the best place) that the action's parameters that form the payload references mainly to the event output parameters (just a suggestion)

File: docs/modeling/triggers.md [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+"action": {
+	"source": "command-consumer",
+	"payload": {
+		"payloadType": "StartProcessPayload",
+		"variables": {
+			"sender": "${emailFrom}",
+			"recipient": "${emailTo}",
+			"subject": "${emailSubject}"
+		},
+		"name": "Signal_0n91cib",
```
Comment (pmartinezga) [inline docs/modeling/triggers.md]:
```suggestion
		"name": "Signal_0n91cib"
```

**Review by pmartinezga — CHANGES_REQUESTED** [review]
Nice work! Few minor things and some suggestions

File: docs/modeling/connectors/README.md (hunk 1 above)
Comment (m-hulbert) [reply docs/modeling/connectors/README.md]: Updated - nice spot.

File: docs/modeling/connectors/README.md (hunk 2 above)
Comment (m-hulbert) [reply docs/modeling/connectors/README.md]: Ah yes. I updated in the triggers but missed it in connectors.

File: docs/modeling/triggers.md (hunk 3 above)
Comment (m-hulbert) [reply docs/modeling/triggers.md]: This is a good point. I've re-worded the `source` property.

File: docs/modeling/triggers.md (hunk 4 above)
Comment (m-hulbert) [reply docs/modeling/triggers.md:67]: Added a line about this for triggers and connector events.

**Review by pmartinezga — APPROVED** [review]

**Pull request merged**